sdk/monitor/monitor-opentelemetry/src/logs/logRecordProcessor.ts

```diff
@@ -15,7 +18,12 @@ export class AzureLogRecordProcessor implements LogRecordProcessor {
     this._metricHandler = metricHandler;
   }
 
-  public onEmit(logRecord: LogRecord): void {
+  public onEmit(logRecord: LogRecord, context: Context): void {
+    const parentSpan = trace.getSpan(context);
+    if (parentSpan && "name" in parentSpan) {
+      // If the parent span has a name we can assume it's a ReadableSpan and cast it as such
+      logRecord.attributes[AI_OPERATION_NAME] = (parentSpan as unknown as ReadableSpan).name;
```

Comment:
Do we want to add this property all the time? I can expect some people do not want this data in there and it would be there for all logs now.

Reply:
The other option I see is if we, like Java, used the `KnownContextTagKeys.AiOperationName` value to set this value on the correlated logs instead of the span name, such that customers using the distro would need to manually set that value in order to get a correlated `Operation Name` value on their logs.

Reply:
Removed edits to the distro and kept the logic in the exporter so that customers can add Span and LogRecord processors that will allow for propagation of the operation name.